Modules/xSharePoint/DSCResources/MSFT_xSPVisioServiceApp/MSFT_xSPVisioServiceApp.psm1

```diff
@@ -0,0 +1,93 @@
+        $params = $args[0]
+        
+        $serviceApps = Get-SPServiceApplication -Name $params.Name -ErrorAction SilentlyContinue 
+        if ($null -eq $serviceApps) { 
+            return $null 
+        }
+        $serviceApp = $serviceApps | Where-Object { $_.TypeName -eq "Visio Graphics Service Application" }
+
+        If ($null -eq $serviceApp) { 
+            return $null 
+        } else {
+            $returnVal =  @{
+                Name = $serviceApp.DisplayName
+                ApplicationPool = $serviceApp.ApplicationPool.Name
+            }
+            return $returnVal
+        }
+    }
+    return $result
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]  [System.String] $Name,
+        [parameter(Mandatory = $true)]  [System.String] $ApplicationPool,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    $result = Get-TargetResource @PSBoundParameters
+
+    if ($result.Count -eq 0) { 
+        Write-Verbose -Message "Creating Visio Graphics Service Application $Name"
+        Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+            $params = $args[0]
+        
+
+        New-SPVisioServiceApplication -Name $params.Name `
+                                      -ApplicationPool $params.ApplicationPool
+        }
+    }
+    else {
+        if ($ApplicationPool -ne $result.ApplicationPool) {
+            Write-Verbose -Message "Updating Visio Graphics Service Application $Name"
+            Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+                $params = $args[0]               
+
+                $appPool = Get-SPServiceApplicationPool -Identity $params.ApplicationPool
+
+                Get-SPServiceApplication -Name $params.Name `
+                    | Where-Object { $_.TypeName -eq "Visio Graphics Service Application" } `
+                    | Set-SPVisioServiceApplication -ServiceApplicationPool $appPool
+            }
+        }
+    }
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [parameter(Mandatory = $true)]  [System.String] $Name,
+        [parameter(Mandatory = $true)]  [System.String] $ApplicationPool,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+    
+    Write-Verbose -Message "Testing for Visio Graphics Service Application '$Name'"
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+    
+    if ($null -eq $CurrentValues) { return $false }
+    return Test-xSharePointSpecificParameters -CurrentValues $CurrentValues -DesiredValues $PSBoundParameters -ValuesToCheck @("ApplicationPool")
+}
+
+Export-ModuleMember -Function *-TargetResource
```

Comment:
Can you add an empty line to the end of the file here? GitHub likes it if files end with an empty line :)